### ISSUE TITLE
Sass compiler @at-root support

### ISSUE THREAD

**Issue by vaadin-bot** [vaadin/sass-compiler#167]
**Originally by @jouni**

___
Sass 3.3 has a new @-rule that makes it easier to control selector nesting, called at-root.

This would significantly help us in implementing custom font support, including font icon support, so that @font-face rules could be included from anywhere in the compilation, not just at the root level – which is usually the case because the theme is wrapped in the `.theme-name` selector.

Otherwise very forced to create some sort of convention how custom fonts are loaded in our themes and in extending themes.

In a nutshell:
```
@mixin custom-font {
  @at-root {
    @font-face {
      src: ...;
      font-family: my-custom-font;
    }
  }
}

.my-theme {
  .some-selector {
    ...
  }

  @include custom-font;
}
```

Output:
```
.my-theme .some-selector {
  ...
}

@font-face {
  src: ...;
  font-family: my-custom-font;
}
```

Blog post describing the @at-root rule in more detail: http://blog.unakravets.com/post/64113156740/sass-3-3-at-root-bem

___
 Imported from https://dev.vaadin.com/ issue #13296

**Comment by SomMeri** [comment]
Do you have plans to implement this? It would help us too.
